src/components/CustomerQuota/CheckoutSuccess/CheckoutSuccessCard.tsx

```diff
@@ -40,22 +48,109 @@ const UsageQuotaTitle: FunctionComponent<{
+      transactions: Transaction[];
+      order: number;
+    };
+  } = {};
+  sortedTransactions?.forEach(item => {
+    const policy = allProducts?.find(
+      policy => policy.category === item.category
+    );
+    const categoryName = policy?.name ?? item.category;
+    const transactionTimeInSeconds = String(
+      Math.floor(item.transactionTime.getTime() / 1000)
+    );
+
+    if (!(transactionTimeInSeconds in transactionsByTimeMap)) {
+      transactionsByTimeMap[transactionTimeInSeconds] = {
+        transactionTime: item.transactionTime,
+        transactions: [],
+        order: -transactionTimeInSeconds
+      };
+    }
+    transactionsByTimeMap[transactionTimeInSeconds].transactions.push({
+      header: categoryName,
+      details: getIdentifierInputDisplay(item.identifierInputs ?? []),
+      quantity: formatQuantityText(
+        item.quantity,
+        policy?.quantity.unit || { type: "POSTFIX", label: " qty" }
+      ),
+      isAppeal: policy?.categoryType === "APPEAL",
+      order: policy?.order ?? BIG_NUMBER
+    });
+  });
+  return transactionsByTimeMap;
+};
+
+export const sortTransactions = (
+  transactionsByTimeMap: TransactionsByTimeMap
+): TransactionsGroup[] => {
+  return Object.entries(transactionsByTimeMap)
+    .sort(([, a], [, b]) => sortTransactionsByOrder(a, b))
+    .map(([, { transactionTime, transactions, order }]) => ({
+      header: format(transactionTime.getTime(), "d MMM yyyy, h:mma"),
+      transactions: transactions.sort(sortTransactionsByOrder),
+      order
+    }));
+};
+
 export const CheckoutSuccessCard: FunctionComponent<CheckoutSuccessCard> = ({
   ids,
   onCancel,
-  checkoutResult,
   quotaResponse
 }) => {
-  const checkoutQuantities = getPurchasedQuantitiesByItem(ids, checkoutResult!);
+  const [isShowFullList, setIsShowFullList] = useState<boolean>(false);
+
   const { getProduct } = useContext(ProductContext);
-  const productType = getProduct(checkoutQuantities[0].category)?.type;
+  const { policies: allProducts } = useContext(CampaignConfigContext);
+  const { sessionToken, endpoint } = useContext(AuthContext);
+  const { pastTransactionsResult } = usePastTransaction(
+    ids,
+    sessionToken,
+    endpoint
+  );
+  // Assumes results are already sorted (valid assumption for results from /transactions/history)
+  const sortedTransactions = pastTransactionsResult;
+
+  const transactionsByTimeMap = groupTransactionsByTime(
+    sortedTransactions,
+    allProducts || []
+  );
+  const transactionsByTimeList = sortTransactions(transactionsByTimeMap);
+
+  const productType =
+    (allProducts && getProduct(allProducts[0].category)?.type) || "REDEEM";
   const { title, description, ctaButtonText } = getCheckoutMessages(
     productType
   );
 
   const showGlobalQuota: boolean =
     !!quotaResponse?.globalQuota &&
-    !!getProduct(checkoutQuantities[0].category)?.quantity.usage;
+    !!sortedTransactions &&
+    sortedTransactions.length > 0 &&
```

Comment:
Are we able to do `sortedTransactions?.length > 0` instead?

Reply:
Hmm then there is a lint error saying "Object is possibly undefined", so I added in the !! check

Reply:
hmm ok, interesting, thought it'd work.